applications/sdp/mspi/src/hrt/hrt.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "hrt.h"
+#include <hal/nrf_vpr_csr_vio.h>
+#include <hal/nrf_vpr_csr_vtim.h>
+
+#define CLK_FIRST_CYCLE_MULTIPLICATOR (3)
+
+void write_single_by_word(volatile struct hrt_ll_xfer xfer_ll_params)
+{
+
+	NRFX_ASSERT(xfer_ll_params.word_size <= MAX_WORD_SIZE);
+	/* Configuration step */
+	uint16_t dir = nrf_vpr_csr_vio_dir_get();
```

Comment:
variables should all be declared at the start of the function, not part way through